project/DistributionPackage.scala

```diff
@@ -261,9 +323,94 @@ object DistributionPackage {
         }
       }
 
+      archive
+    }
+
+    private def copyGraal(
+      os: OS,
+      architecture: Architecture,
+      runtimeDir: File
+    ): Unit = {
+      val archive = graalArchive(os, architecture)
       extract(archive, runtimeDir)
     }
 
+    /** Prepare the GraalVM package.
+      *
+      * @param log the logger
+      * @param os the system type
+      * @param architecture the architecture type
+      * @return the path to the created GraalVM package
+      */
+    def createGraalPackage(
+      log: ManagedLogger,
+      os: OS,
+      architecture: Architecture
+    ): File = {
+      log.info("Building GraalVM distribution")
+      val archive = downloadGraal(log, os, architecture)
+
+      if (os != OS.Windows) {
```

Comment:
The necessary condition for this isn't really `isWindows`. It's a property of windows that it doesn't support sulong, so I would add an OS property `hasSupportForSulong` and guard this on that instead. It's a better expression of what's actually going on.

Reply:
Updated :+1: